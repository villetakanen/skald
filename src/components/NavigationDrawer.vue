<template>
  <v-list>
    <template v-if="siteID !== null">
      <v-subheader>Site</v-subheader>
      <v-list-item :to="'/v/'+siteID">
        <v-list-item-action><v-icon>mdi-dice-d20</v-icon></v-list-item-action>

        <v-list-item-title >Site Home</v-list-item-title>
      </v-list-item>
      <v-list-item :to="'/c/site/'+siteID">
        <v-list-item-action><v-icon>mdi-dice-d4</v-icon></v-list-item-action>

        <v-list-item-title >Settings</v-list-item-title>
      </v-list-item>
    </template>

    <v-subheader>Meta</v-subheader>
    <v-list-item :to="'/'">
      <v-list-item-action><v-icon>mdi-home</v-icon></v-list-item-action>

      <v-list-item-title >Home</v-list-item-title>
    </v-list-item>

    <v-list-item :to="'/l/sites'">
      <v-list-item-action><v-icon>mdi-view-dashboard</v-icon></v-list-item-action>

      <v-list-item-title>All sites</v-list-item-title>
    </v-list-item>

    <v-list-item :to="'/v/skald/beta-about'">
      <v-list-item-action><v-icon>mdi-information-outline</v-icon></v-list-item-action>

      <v-list-item-title >About</v-list-item-title>
    </v-list-item>
    <v-list-item href="https://github.com/vitku/skald">
      <v-list-item-action><v-icon>mdi-language-javascript</v-icon></v-list-item-action>
      <v-list-item-title>GitHub</v-list-item-title>
    </v-list-item>
    <v-list-item :to="'/v/skald/beta-backlog'">
      <v-list-item-action><v-icon>mdi-cup-water</v-icon></v-list-item-action>

      <v-list-item-title >{{version}}</v-list-item-title>
    </v-list-item>
    <v-list-item :to="'/stylebook'">
      <v-list-item-action><v-icon>mdi-fridge</v-icon></v-list-item-action>

      <v-list-item-title >Stylebook</v-list-item-title>
    </v-list-item>
  </v-list>
</template>

<script>
export default {
  computed: {
    version () {
      return this.$store.getters['version']()
    },
    siteID () {
      return this.$store.getters['binder/siteID']()
    }
  }
}
</script>
