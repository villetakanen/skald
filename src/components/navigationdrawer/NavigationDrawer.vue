<template>
  <v-list>
    <template v-if="siteid">
      <v-list-item :to="`/v/${siteid}`">
        <v-list-item-action><v-icon>mdi-dice-3</v-icon></v-list-item-action>
        <v-list-item-title >{{siteName}}</v-list-item-title>
      </v-list-item>
      <v-list-item
        id="navi-site-in-game-secrets-link"
        :to="`/g/${siteid}/gm/players`">
        <v-list-item-action><v-icon>mdi-account-multiple</v-icon></v-list-item-action>
        <v-list-item-title >{{$t('view_gm_players.link_to')}}</v-list-item-title>
      </v-list-item>
      <v-list-item id="nav-page-list-link" :to="'/l/pages/'+siteid">
        <v-list-item-action><v-icon>mdi-file-tree</v-icon></v-list-item-action>
        <v-list-item-title >{{$t('navigation_to_pages')}}</v-list-item-title>
      </v-list-item>
      <!-- @click: CreatePageCard, the dialog for adding a page -->
      <!-- AddPageListItem v-if="editorActions" -->
      <CreatePageAction v-if="editorActions"/>
      <v-list-item id="navi-attachment-list-link" :to="'/l/attachments/'+siteid">
        <v-list-item-action><v-icon>mdi-view-grid-plus</v-icon></v-list-item-action>
        <v-list-item-title >{{$t('navigation_to_attachments')}}</v-list-item-title>
      </v-list-item>

      <!-- Site settings -->
      <v-list-item
        id="navi-site-settings-link"
        v-if="editorActions"
        :to="'/c/site/'+siteid">
        <v-list-item-action><v-icon>mdi-cog-box</v-icon></v-list-item-action>
        <v-list-item-title >{{$t('navigation_to_site_settings')}}</v-list-item-title>
      </v-list-item>

      <v-subheader>Meta</v-subheader>
    </template>
    <v-list-item :to="'/'">
      <v-list-item-action><v-icon>mdi-home</v-icon></v-list-item-action>
      <v-list-item-title >{{$t('navigation_to_skald_home')}}</v-list-item-title>
    </v-list-item>
    <v-list-item :to="'/l/sites'">
      <v-list-item-action><v-icon>mdi-view-dashboard</v-icon></v-list-item-action>
      <v-list-item-title>{{$t('navigation_to_site_index')}}</v-list-item-title>
    </v-list-item>
    <v-list-item href="https://github.com/vitku/skald">
      <v-list-item-action><v-icon>mdi-language-javascript</v-icon></v-list-item-action>
      <v-list-item-title>GitHub</v-list-item-title>
    </v-list-item>

    <v-divider></v-divider>

    <LanguageSwitcher/>

  </v-list>
</template>

<script lang="ts">
import Vue from 'vue'
import VueCompositionApi, { defineComponent, computed } from '@vue/composition-api'
import AddPageListItem from '@/components/AddPageListItem.vue'
import LanguageSwitcher from '@/components/LanguageSwitcher.vue'
import CreatePageAction from '@/components/createpageaction/CreatePageAction.vue'
import { useSite } from '@/lib/useSite'
import { useProfile } from '@/lib/useProfile'
Vue.use(VueCompositionApi)

export default defineComponent({
  components: {
    AddPageListItem,
    LanguageSwitcher,
    CreatePageAction
  },
  setup () {
    const { site } = useSite()
    const siteid = computed(() => site.value.siteid)
    const siteName = computed(() => site.value.name)
    const { activeProfile } = useProfile()
    const editorActions = computed(() =>
      activeProfile.value &&
      activeProfile.value.owns &&
      activeProfile.value.owns.includes(site.value.siteid))
    return { siteid, siteName, editorActions }
  }
})
</script>
