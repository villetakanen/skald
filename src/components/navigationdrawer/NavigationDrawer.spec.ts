import { shallowMount } from '@vue/test-utils'
import NavigationDrawer from './NavigationDrawer.vue'

describe('Mounted App', () => {
  const wrapper = shallowMount(NavigationDrawer)

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
