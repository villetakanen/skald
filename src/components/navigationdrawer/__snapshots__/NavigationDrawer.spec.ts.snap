// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mounted App renders correctly 1`] = `
<v-list-stub
  tag="div"
>
  <!---->
   
  <v-list-item-stub
    activeclass=""
    tag="div"
    to="/"
  >
    <v-list-item-action-stub>
      <v-icon-stub>
        mdi-home
      </v-icon-stub>
    </v-list-item-action-stub>
     
    <v-list-item-title-stub>
      
    </v-list-item-title-stub>
  </v-list-item-stub>
   
  <v-list-item-stub
    activeclass=""
    tag="div"
    to="/l/sites"
  >
    <v-list-item-action-stub>
      <v-icon-stub>
        mdi-view-dashboard
      </v-icon-stub>
    </v-list-item-action-stub>
     
    <v-list-item-title-stub>
      
    </v-list-item-title-stub>
  </v-list-item-stub>
   
  <v-list-item-stub
    activeclass=""
    href="https://github.com/vitku/skald"
    tag="div"
  >
    <v-list-item-action-stub>
      <v-icon-stub>
        mdi-language-javascript
      </v-icon-stub>
    </v-list-item-action-stub>
     
    <v-list-item-title-stub>
      GitHub
    </v-list-item-title-stub>
  </v-list-item-stub>
   
  <v-divider-stub />
   
  <languageswitcher-stub />
</v-list-stub>
`;
